src/services/users.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Service } from 'typedi';
+import { compareSync, genSaltSync, hashSync } from 'bcrypt';
+import { getRepository } from 'typeorm';
+import { User } from '@entities/user.entity';
+import { JWTService } from '@services/jwt.service';
+import { Errors } from '@constants/errorMessages';
+
+@Service()
```

Comment:
Nice 😎 , I'd (personally) make some changes to the service, I'm not a huge fan of services like `UserService` as IMHO a service should have one specific goal, for example, the sign-in of a user.

Moreover, the methods such as `index, post, put, delete show` I think that could be removed from the service as they don't really have much logic for themselves, they are one-line methods that do the job needed so they (again, IMO) could be in the controller.